### DISCUSSION TITLE
Running Bash with Box86 & Box64 and x86-64 Wine ?

### DISCUSSION THREAD

**Discussion by xDoge26** [ptitSeb/box86#719] [Q&A] [answered]
Do i need to compile box first or just git clone and export BOX64_BASH=~/box64/tests/bash ?
And wine setup should be the same right ?
```
sudo ln -s ~/wine/bin/wine /usr/local/bin/wine
sudo ln -s ~/wine/bin/wineserver /usr/local/bin/wineserver
sudo ln -s ~/wine/bine/wineboot /usr/local/bin/wineboot
sudo ln -s ~/wine/bin/wine64 /usr/local/bin/wine64

**Comment by ptitSeb** [comment]
You can do the export when ever you like. Just make sure the file is there when you run box.
Wine setup looks good yes.

**Reply by xDoge26** [reply]
> You can do the export when ever you like. Just make sure the file is there when you run box. Wine setup looks good yes.

Then can i use Box86-64 Bash without compiling it ?

If i want to run 64bit wine in box64 bash , what command should be : `box64 wine *.exe` or `box64 wine64 *.exe`?
And with 32bit wine is ` box86 wine *.exe ?

Do i need both wine i386 and wine amd64 or just amd64 alone ? , and box86-64 bash only avaiable in box86 ver 0.2.8 and box64 ver 2.0 right ?

**Comment by ptitSeb** [comment]
Use without compiling what? bash is a binary here, no need to compile it.
box86/box64 are source only here, you need to build it and install it before using it.

Wine amd64 is enough, and will be compatible with both 64bits and 32bits binary. If you use Wine 32 bits, 64bits binary will not be usable (also, you need seperate WINEPREFIX if you want to use both versions of Wine).

Yes, you need current version of box86 & box64 to use bash. box86 v0.2.8 and box64 v0.2.0 are not out yet (they will be soon).

**Reply by xDoge26** [reply]
> Use without compiling what? bash is a binary here, no need to compile it. box86/box64 are source only here, you need to build it and install it before using it.
> 
> Wine amd64 is enough, and will be compatible with both 64bits and 32bits binary. If you use Wine 32 bits, 64bits binary will not be usable (also, you need seperate WINEPREFIX if you want to use both versions of Wine).
> 
> Yes, you need current version of box86 & box64 to use bash. box86 v0.2.8 and box64 v0.2.0 are not out yet (they will be soon).

so when box86 0.2.8 arrive . i can use bash + box86 with 64bit wine prefix and use 32bit binary included in wine amd64 right ?

**Reply by ptitSeb** [reply]
You can already if you build from current sources of box86 & box64.

**Reply by xDoge26** [reply]
> You can already if you build from current sources of box86 & box64.

i already compiled box 86-64 with git clone . Which package i need , to use box86-64 +wine in bash without binfmt

![Screenshot_2022-11-09-21-52-12-730-edit_ru zdevs zarchiver](https://user-images.githubusercontent.com/92904701/200863522-47a60e6f-1c38-43c1-9738-ba625adf383c.jpg)

**Reply by ptitSeb** [reply]
Don't use a deb, you need to extract stuff by hand, it's not easy.
Use a tgz, like thise one: https://www.playonlinux.com/wine/binaries/phoenicis/upstream-linux-amd64/PlayOnLinux-wine-6.0.1-upstream-linux-amd64.tar.gz

**Reply by xDoge26** [reply]
> Don't use a deb, you need to extract stuff by hand, it's not easy. Use a tgz, like thise one: https://www.playonlinux.com/wine/binaries/phoenicis/upstream-linux-amd64/PlayOnLinux-wine-6.0.1-upstream-linux-amd64.tar.gz

im just wanna ask can box86 use `#wine64 support files (required for wine64 / can work alongside wine_i386 main bin) ` without binfmt ,or i have to use `#wine_i386 support files (required for wine_i386 if no wine64 / CONFLICTS WITH wine64 support files)`

**Reply by ptitSeb** [reply]
the deb wine64 is 64bits only and needs box64.
But again, just use the tgz I pointed, it's all setup with both wine and wine64 ready to use.

**Reply by xDoge26** [reply]
> the deb wine64 is 64bits only and needs box64. But again, just use the tgz I pointed, it's all setup with both wine and wine64 ready to use.

do you know why cant create 64bit wine prefix with box64 when using amd64 wine from winehq ?

**Reply by ptitSeb** [reply]
latest version of wine use 64->32bits mecanism that is not yet supported by box64.

**Reply by ptitSeb** [reply]
Also, wine 64bits deb package are only the 64bits binary, but to run 7.0 version of wine 64bits, you also need 32bits parts.
You can download versions there also: https://github.com/Kron4ek/Wine-Builds/releases
I tried 7.5 64bits and it worked fine.

**Reply by xDoge26** [reply]
> Also, wine 64bits deb package are only the 64bits binary, but to run 7.0 version of wine 64bits, you also need 32bits parts. You can download versions there also: https://github.com/Kron4ek/Wine-Builds/releases I tried 7.5 64bits and it worked fine.

does wine from this compatible with box64 ? box86 worked fine with winehq , but box64 does not

**Reply by ptitSeb** [reply]
I took the 7.5, and it worked (but you need both box64 & box86). Recent version like 7.20 will not work for now.

**Comment by ptitSeb** [comment, marked as answer]
Is this a proot or a chroot?
because proot have a different signal handling that interfere with box signal handling, making some things not to work correctly

**Reply by xDoge26** [reply]
> Is this a proot or a chroot? because proot have a different signal handling that interfere with box signal handling, making some things not to work correctly

proot , but box86 work fine in proot , only box64 isnt

**Reply by ptitSeb** [reply]
version 7.5 64bits (not stagging or tkg) from that site works fine on my Phytium ARM miniPC on Debian. So I don't know what wrong, exept it could be a side effect of the proot.
Try to run with `BOX64_LOG=1 BOX64_ROLLING_LOG=1 BOX64_SHOWBT=1 BOX64_SHOWSEGV=1` to see what error log you get (you will need to copy/paste the actual text, a screenshot will probably not work)

**Reply by xDoge26** [reply]
> version 7.5 64bits (not stagging or tkg) from that site works fine on my Phytium ARM miniPC on Debian. So I don't know what wrong, exept it could be a side effect of the proot. Try to run with `BOX64_LOG=1 BOX64_ROLLING_LOG=1 BOX64_SHOWBT=1 BOX64_SHOWSEGV=1` to see what error log you get (you will need to copy/paste the actual text, a screenshot will probably not work)

Here
[text.txt](https://github.com/ptitSeb/box86/files/10001399/text.txt)

**Reply by ptitSeb** [reply]
Ah well, sorry, the error is on the 32bits side it seems. You need to re-run with `BOX86_LOG=1 BOX86_ROLLING_LOG=1 BOX86_SHOWSEGV=1` instead

**Reply by ptitSeb** [reply]
NO, not really.
The 32bits shows a "double free or memory corruption" message. That's bad, but I canoot have any more details here.
The 64bits shows no box error. Wine cannot load `kernel32.dll`, with error `c0000135`. Seems bad. I think it's because some memory address is not available, but it doesn't seems related to box

**Comment by xDoge26** [comment]
Turn out the problem come from Android SELinux not proot Termux , disable it and box64 will create wine prefix like normal
